src/services/impl/FileTextReader.hh

```diff
@@ -0,0 +1,52 @@
+#ifndef RA_SERVICES_FILETEXTREADER
+#define RA_SERVICES_FILETEXTREADER
+#pragma once
+
+#include "services\TextReader.hh"
+
+#include "RA_StringUtils.h"
+
+#include <fstream>
+
+namespace ra {
+namespace services {
+namespace impl {
+
+class FileTextReader : public ra::services::TextReader
+{
+public:
+    FileTextReader(const std::wstring& sFilename)
+        : m_iStream(sFilename)
+    {
+    }
+
+    bool GetLine(_Out_ std::string& sLine) override
+    {
+        if (!std::getline(m_iStream, sLine))
+            return false;
+
+        ra::TrimLineEnding(sLine);
+        return true;
+    }
+
+    bool GetLine(_Out_ std::wstring& sLine) override
+    {
+        std::string sNarrowLine;
+        if (!std::getline(m_iStream, sNarrowLine))
+            return false;
+
+        sLine = ra::Widen(ra::TrimLineEnding(sNarrowLine));
+        return true;
+    }
+
+    std::ifstream& GetFStream() { return m_iStream; }
```

Comment:
Is this meant to be modified? If not, mark it `const `and `noexcept`.

Reply:
you cannot read from a const stream. the position gets updated.

in this particular case, it's exposed for consumption by the rapidjson helper function `LoadDocument`. it's not actually part of the interface.